examples/Coffee_Machine_FSM/coffee.py

```diff
@@ -0,0 +1,201 @@
+from enum import Enum
+from typing import Union, List
+
+
+
+class StatusCode(Enum):
+    OK = 1
+    ERROR = 0
+    COFFEE_PRODUCED = 2
+
+
+
+class State:
+    def __init__(self, name: str, coffee_machine: "CoffeeMachine") -> None:
```

Comment:
Why does the state own a coffee machine? I would rather expect the coffee machine having a state. I think, this can be overcome by returning new states in the handle functions of the states.